include/board.h

```diff
@@ -12,24 +12,22 @@ class Piece;
 
 class Board {
     // https://www.chess.com/terms/fen-chess
-    std::vector<std::vector<std::shared_ptr<Piece>>> board;
+    std::vector<std::vector<std::shared_ptr<Piece>>> grid;
     Color active_color;
-    std::map<Color, std::map<CastleSide, bool>> castle_rights;
-    std::set<Coordinate> en_passant_targets;
-    int halfmove_clock;
-    int fullmove_clock;
     std::set<std::shared_ptr<Piece>> white_pieces;
     std::set<std::shared_ptr<Piece>> black_pieces;
     std::shared_ptr<Piece> white_king;
     std::shared_ptr<Piece> black_king;
+    std::map<Color, std::map<CastleSide, bool>> castle_rights;
+    std::set<Coordinate> en_passant_targets;
+    int halfmove_clock;
+    int fullmove_clock;
     bool verify_board();
-    void place_piece(
-        Color color, Coordinate square, PieceType type);
-    void remove_piece(Coordinate square);
     std::string deserialize();
 
 public:
     Board();
+    Board(int i);
```

Comment:
the board constructor with (int i) should be merged into the default constructor